src/GraphQLController.php

```diff
@@ -86,6 +98,43 @@ protected function queryContext(string $query, ?array $params, string $schema)
         }
     }
 
+    /**
+     * @param  string  $schema
+     * @param  array  $input
+     * @return string
+     * @throws Error\AutomaticPersistedQueriesError
+     * @throws \Psr\SimpleCache\InvalidArgumentException
+     */
+    protected function getQueryFromAPQIfNeeded(string $schema, array $input): string
+    {
+        $query = $input['query'] ?? ''; // NOTE: why ?
+
+        $persistedQueryInput = data_get($input, 'extensions.persistedQuery');
+        if ($persistedQueryInput && (! config('graphql.apq.enable'))) {
+            throw Error\AutomaticPersistedQueriesError::persistedQueriesNotSupported();
+        }
+
+        if (null !== ($hash = data_get($persistedQueryInput, 'sha256Hash'))) {
+            $apqCacheDriver = config('graphql.apq.cache_driver');
+            $apqCachePrefix = config('graphql.apq.cache_prefix');
+            $apqCacheIdentifier = implode('.', [$apqCachePrefix, $hash]);
+
+            if (! array_key_exists('query', $input)) {
+                if (! cache()->has($apqCacheIdentifier)) {
+                    throw Error\AutomaticPersistedQueriesError::persistedQueriesNotFound();
+                }
+                $query = cache()->driver($apqCacheDriver)->get($apqCacheIdentifier);
+            } else {
+                if ($hash !== hash('sha256', $query)) {
+                    throw Error\AutomaticPersistedQueriesError::internalServerError('provided sha does not match query');
```

Comment:
I know...it's not "nice" but I kept the same as of apollo-server-express
https://github.com/apollographql/apollo-server/blob/ad4c3acfaff1bfe626f2ba64e2d0af00c55ab5a2/packages/apollo-server-core/src/requestPipeline.ts#L213

My thinking is maybe it's best to keep it aligned even if I don't think it can cause problems.

Reply:
ah, that makes sense!

I would be nice then to document the reasoning for this, otherwise next week I get a PR from someone wanting to change this maybe…

IMHO a direct reference to `apollo-server` like you did in the comment is fine too.

Very thoughtful decision about this, thanks!

Reply:
PS: using a constant in your already nice error class would also be fine 🤷‍♀️

Reply:
Ok